tests/integration_tests/functional/test_max_vcpus.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests scenario for microvms with max vcpus(32)."""
+import os
+import select
+from framework.state_machine import TestState
+
+MAX_VCPUS = 32
+
+
+class WaitCPUs(TestState):
+    """Initial state when we wait for the login prompt."""
+
+    def handle_input(self, microvm, input_char) -> TestState:
+        """Handle input and return next state."""
+        if self.match(input_char):
+            return TestFinished()
+        return self
+
+
+class TestFinished(TestState):
+    """Test complete and successful."""
+
+    def handle_input(self, microvm, input_char) -> TestState:
+        """Return self since the test is about to end."""
+        return self
+
+
+def test_max_vcpus(test_microvm_with_ssh):
+    """Test if all configured vcpus are brought up at boot."""
+    microvm = test_microvm_with_ssh
+    microvm.jailer.daemonize = False
+    microvm.spawn()
+
+    # Configure a microVM with 32 vCPUs and a serial console.
+    microvm.basic_config(vcpu_count=MAX_VCPUS,
+                         boot_args='console=ttyS0 reboot=k panic=1 pci=off')
+
+    microvm.start()
+
+    # Setup screen log
+    screen_log = "/tmp/screen.log"
+    screen_log_fd = os.open(screen_log, os.O_RDONLY)
+
+    poller = select.poll()
+
+    # Set initial state - wait for cpus boot string
+    current_state = WaitCPUs("{} processors activated".format(MAX_VCPUS))
+
+    poller.register(screen_log_fd, select.POLLIN | select.POLLHUP)
+
+    while not isinstance(current_state, TestFinished):
```

Comment:
I might be missing something, but can this loop forever in case the CPUs don't become available?

Reply:
That's a fair point, but the test should timeout after 300s and fail.